presto-docs/src/main/sphinx/functions/ml.rst

```diff
@@ -0,0 +1,151 @@
+===================
+Machine Learning Functions
+===================
+
+The machine learning plugin in Presto provides machine learning functionality
+as an aggregation function. It enables us to train Support Vector Machine (SVM)
+based classifier and regressor for the supervised learning problems.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+.. note::
+  These machine learning functions are not fully optimized in a distributed manner.
+  Since the final training process is executed in a single instance, it has a limited capability to train the enormous scale data for now.
+
+Feature Vector
+--------------
+
+To solve a problem with the machine learning technique especially as a
+supervised learning problem, it is necessary to represent the data set
+with the sequence of pairs of labels and feature vector. A label is a
+target value you want to predict from the unseen feature and a feature is a
+An N-dimensional vector whose elements are numerical values. In Presto, feature
+vector is represented as a map type value whose key is an index of each
+feature so that it is capable of expressing a sparse vector.
+Since classifiers and regressors in Presto can recognize the
+map type feature vector, there is a function to construct the feature
+from the existing numerical values, :func:`features`::
+
+    SELECT features(1.0, 2.0, 3.0) features;
+
+.. code-block:: none
+
+           features
+    -----------------------
+     {0=1.0, 1=2.0, 2=3.0}
+
+The output from :func:`features` can be directly passed to ML functions.
+
+Classification
+---------------
+
+Classification is a type of supervised learning problem to predict the distinct
+label from the given feature vector. The interface looks close to what in
+Teradata Aster or `BigQuery ML <https://cloud.google.com/bigquery-ml/docs/bigqueryml-intro>`_. :func:`learn_classifier` constructs the SVM
+model from the sequence of pairs of labels and features::
+
+    SELECT
+      learn_classifier(species, features(sepal_length, sepal_width, petal_length, petal_width))
+    FROM
+      iris
+
+It returns the trained model in a serialized format.
+
+.. code-block:: none
+
+                          model
+    -------------------------------------------------
+     3c 43 6c 61 73 73 69 66 69 65 72 28 76 61 72 63
+     68 61 72 29 3e
+
+:func:`classify` returns the predicted label by using the trained model. For now, we
```

Comment:
Dont use "For now" and "we"

The trained model can not be saved natively, and needs to be passed in the format of a nested query::